src/Models/ProductModel.php

```diff
@@ -50,8 +47,9 @@ public function addProduct(ProductEntityInterface $productEntity): bool
     public function checkProductExists(string $sku): bool
     {
         $query = $this->db->prepare("SELECT `id` FROM `products` WHERE `sku` = ?");
-        $result = $query->execute([$sku]);
+        $query->execute([$sku]);
+        $result = $query->fetch();
 
-        return mysqli_num_rows($result) !== 0;
+        return $result > 0;
```

Comment:
$result will either be an array if it exists or false if it doesn't. This seems to work but it reads a bit weird imo.